OpenTrons/opentrons-api
# feat(protocol-designer, step-generation): waste chute commands for dispense, air_gap, and blow_out

closes RAUT-764, RAUT-812, RAUT-855

# Overview

This PR adds waste chute as an option in the Transfer form's dropdown to dispense and blow out into. This involved changing the utilities a bit that depend on labware definitions and updating `liquidState` to include `additionalEquipment` key to keep track of waste chute's liquid state. Then wiring up the atomic commands in `transfer` and `consolidate`

<img width="931" alt="Screen Shot 2023-10-27 at 4 38 05 PM" src="https://github.com/Opentrons/opentrons/assets/66035149/008e189c-95f6-4e2b-a0a8-fde36bc05053">

# Test Plan

Turn on deck modification ff and add a waste chute to the Flex protocol. 

1. Create a transfer step (1 to 1 wells) and select `waste chute` for dispensing. Look at the advanced settings and see that `mix`, `touch_tip`, and `tip positioning` are disabled and the `well order` is not there. Click on the `blow_out` checkbox and see that `waste chute` is an option in the dropdown. Create the step and see that there are no timeline errors. Look at redux and examine the command's liquid state. See that the waste chute entity exists in `additionalEquipment` key and that the proper volume is added. Also add all the advanced settings for dispense (air gap and blow out).
2. Now, create another transfer step, this time do a consolidate so multiple to 1 wells. Make sure that the waste chute option matches what you previously saw and that you can create the step okay and nothing errors. Examine the liquid state and see that it is correct.
3. download the protocol and see that the correct commands are emitted: `moveToAddressableArea` followed by `dispenseInPlace`, `blowOutInPlace` or `aspirateInPlace`

NOTE: since the waste chute can only be dispensed/blow_out into and its only 1 _fake well_, only transfer and consolidate options are available. Mix and distribute are not available.

4. Finally, create a protocol with a multichannel and either add a transfer or consolidate. See that nothing errors and the liquid state updates accordingly.

# Changelog

- lots of refactoring of utils so they don't rely on labware definition and stub in A1 for a well
- extend `liquidState` key to include `additionalEquipment` which is only the waste chute (we could potentially rename this) so we can keep track of waste chute's liquid state.
- add waste chute as an option in the drop down menus 
- add disabled + tooltips to mix and touch tip, plus refactor/clean up the previous code for touch tip not being allowed in certain labware due to the quirk
- wire `aspirateInPlace` command into `wasteChuteCommandsUtil`
- create some utils to grab the correct commands in step-generation for interacting with the waste chute
- go through `transfer` and `consolidate` and stub in the correct commands
- add a bit of test coverage

# Review requests

Does it make sense to add `additionalEquipment` to the `liquidState` key in `RobotState`? Since Brian mentioned that tracking liquid in the waste chute would be nice to have. I should for sure extend `liquidState` since waste chute is not a labware but is `additionalEquipment` the appropriate name since it only encompasses the waste chute (for now)? 

# Risk assessment

low
